src/web/context/GlobalContext.tsx

```diff
@@ -0,0 +1,231 @@
+ * @param {Action} action Contains data associated with a state update 
+ * @returns {GlobalContextType} A copy of the updated state
+ */
+function globalReducer(state: GlobalContextType, action: Action) {
+    let mutableState = {...state}
+    switch (action.type) {
+        case 'SAVED_CLIENT_ID':
+            return handleSavedClientID(mutableState, action.clientID)
+
+        case 'TAKE_CONTROL_SUCCESS':
+            return handleTakeControlSuccess(mutableState)
+            
+        case 'EXITED_RC_MODE':
+            return handleExitedRCMode(mutableState)
+
+        case 'CLOSED_HUB_DETAILS':
+            return handleClosedHubDetails(mutableState)
+        
+        case 'CLICKED_HUB_TAB':
+            return handleClickedHubTab(mutableState)
+
+        case 'CLICKED_BOT_TAB':
+            return handleClickedBotTab(mutableState)
+
+        case 'CLICKED_HUB_ACCORDION':
+            return handleClickedHubAccordion(mutableState, action.hubAccordionName)
+
+        default:
+            return state
+    }
+}
+
+/**
+ * Adds the client ID to state
+ * 
+ * @param {GlobalContextType} mutableState State object ref for making modifications
+ * @param {string} clientID ID associated with the client session
+ * @returns {GlobalContextType} Updated mutable state object
+ */
+function handleSavedClientID(mutableState: GlobalContextType, clientID: string) {
+    mutableState.clientID = clientID
+    return mutableState
+}
+
+/**
+ * Sets the client ID saved in state to be the controlling client ID
+ * 
+ * @param {GlobalContextType} mutableState State object ref for making modifications
+ * @returns {GlobalContextType} Updated mutable state object
+ */
+function handleTakeControlSuccess(mutableState: GlobalContextType) {
+    mutableState.controllingClientID = mutableState.clientID
+    return mutableState
+}
+
+/**
+ * Turns off RC Mode on the client-side
+ * 
+ * @param {GlobalContextType} mutableState State object ref for making modifications
+ * @returns {GlobalContextType} Updated mutable state object
+ */
+function handleExitedRCMode(mutableState: GlobalContextType) {
+    mutableState.isRCMode = false
+    return mutableState
+}
+
+/**
+ * Closes the HubDetails panel
+ * 
+ * @param {GlobalContextType} mutableState State object ref for making modifications
+ * @returns {GlobalContextType} Updated mutable state object
+ */
+function handleClosedHubDetails(mutableState: GlobalContextType) {
+    mutableState.showHubDetails = false
+    return mutableState
+}
+
+/**
+ * Handles the interplay between selecting the hub and clicking the HubTab
+ * 
+ * @param {GlobalContextType} mutableState State object ref for making modifications
+ * @returns {GlobalContextType} Updated mutable state object
+ */
+function handleClickedHubTab(mutableState: GlobalContextType) {
+    const isHubSelected = mutableState.selectedPodElement !== null && mutableState.selectedPodElement.type === PodElement.HUB
```

Comment:
This could be condensed with this using ternary operator:

const isHubSelected = mutableState.selectedPodElement !== null && mutableState.selectedPodElement.type === PodElement.HUB;
mutableState.showHubDetails = !isHubSelected;

Reply:
Readability in your solution is probably better.